MVVMRxSwiftDemoTests/Sources/Specs/Data/Repositories/User/UserSessionRepositorySpec.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  UserSessionRepositorySpec.swift
+//  MVVMRxSwiftDemoTests
+//
+//  Created by Chananchida F. on 7/23/21.
+//
+// swiftlint:disable closure_body_length
+
+@testable import MVVMRxSwiftDemo
+import Quick
+import Nimble
+import RxSwift
+import RxNimble
+
+final class UserSessionRepositorySpec: QuickSpec {
+
+    override func spec() {
+
+        var userSessionRepository: UserSessionRepository!
+        var keychain: KeychainMock!
+
+        describe("a UserSessionRepository") {
+
+            beforeEach {
+                keychain = KeychainMock()
+                userSessionRepository = UserSessionRepository(keychain: keychain)
+            }
+            
+            context("when getIsLoggedIn is called and keychain has isLoggedIn") {
+                
+                beforeEach {
+                    try? keychain.set(true, for: .isLoggedIn)
+                }
+                
+                it("returns correct response") {
+                    expect(
+                        try userSessionRepository.getIsLoggedIn()
+                            .toBlocking(timeout: 1_000)
+                            .last()
+                    )
+                    .to(equal(true))
+                }
+            }
+
+            context("when saveToken is called") {
+                
+                let token = APIToken.dummy
+                let keychainToken = KeychainToken(token: token)
+                
+                beforeEach {
+                    _ = try? userSessionRepository.saveToken(keychainToken).toBlocking(timeout: 1_000).last()
+                }
+                
+                it("keychain has token") {
+                    expect(try keychain.get(.token).hasValue) == true
```

Comment:
I think we can use `notTo(beNil())` instead of `.hasValue`
```suggestion
                    expect(try keychain.get(.token)).notTo(beNil())
```